### PR TITLE
chore: create a github action for publishing packages on npm

### PR DESCRIPTION
**Pull request by m-davoodi**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      # Setup .npmrc file to publish to npm
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+          registry-url: 'https://registry.npmjs.org'
```
Comment (mh-alahdadian) [inline .github/workflows/publish.yml]: use 'https://registry.yarnpkg.com' instead

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1,7 +1,4 @@
-# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
-# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
-
-name: CI/CD
+name: test
```
Comment (mh-alahdadian) [inline .github/workflows/test.yml]: IMO: validate ?

File: package.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "sahab",
+  "name": "@my-sahab/gooni",
```
Comment (mh-alahdadian) [inline package.json:2]: why ? 
this package is root and should not published

**Review by mh-alahdadian — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (m-davoodi) [reply .github/workflows/publish.yml]: no needed

File: package.json (hunk 3 above)
Comment (m-davoodi) [reply package.json:2]: Yes, it's not publishable. But I think this name is better

**Review by mh-alahdadian — APPROVED** [review]

**Pull request merged**